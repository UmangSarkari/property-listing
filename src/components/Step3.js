import React, { useCallback, useState, useMemo } from "react";

import { Button, Typography } from "@material-ui/core";
import makeStyles from "@material-ui/core/styles/makeStyles";
import { useDropzone } from "react-dropzone";
import ImagePreview from "../molecules/ImagePreview";
import PropTypes from "prop-types";

import { STRINGS } from "../strings";

const styles = makeStyles(() => ({
  submitButton: {
    color: "white",
    borderRadius: 8,
    fontSize: 8,
    width: 100,
    margin: "3px 3px",
    padding: "6px 8px",
  },
  button: {
    color: "white",
    borderRadius: 16,
    width: 250,
    margin: "16px 5px",
    padding: "12px 16px",
  },
  error: {
    fontSize: "14px",
    textAlign: "center",
    color: "red",
  },
  step3: {
    marginTop: 20,
    borderTop: "1px dashed",
  },
  dragDrop: {
    padding: 20,
    border: "3px blue dashed",
    maxWidth: 400,
    borderRadius: 20,
  },
  submit: {
    marginTop: 20,
    flexDirection: "row",
  },
}));

const Step3 = ({ data }) => {
  let classes = styles();
  const [images, setImages] = useState([]);
  const [featured, setFeatured] = useState(null);
  const [error, setError] = useState("");

  const onDrop = useCallback((acceptedFiles) => {
    acceptedFiles.forEach((file) => {
      const reader = new FileReader();
      reader.onload = function (e) {
        setImages((prevState) => [
          ...prevState,
          { id: Date.now(), src: e.target.result },
        ]);
      };
      reader.readAsDataURL(file);
    });
  }, []);

  const { getRootProps, getInputProps } = useDropzone({
    onDrop,
    accept: {
      "image/png": [".png", ".jpg", ".jpeg"],
      "image/jpg": [".png", ".jpg", ".jpeg"],
    },
  });
  const showDragDrop = useMemo(() => images.length < 4, [images]);

  const onSubmit = useCallback(() => {
    if (!images.length) {
      setError(STRINGS.SELECT_IMAGE);
      return;
    }
    Object.keys(data).map((key) =>
      console.log(`${key.toUpperCase()} ----> `, data?.[key])
    );
    images.map((image, index) =>
      console.log("Image #", index + 1, " ---->", image.src)
    );
  }, [images, data]);

  const onFeatured = useCallback(
    (e, id) => {
      if (featured === id) {
        setFeatured(null);
      } else {
        setFeatured(id);
      }
    },
    [featured]
  );

  return (
    <div id="step3" className={classes.step3}>
      <ImagePreview
        images={images}
        onFeatured={onFeatured}
        featured={featured}
      />
      {showDragDrop && (
        <div {...getRootProps()} className={classes.dragDrop}>
          <input {...getInputProps()} />
          <p>{STRINGS.DRAG_DROP}</p>
          <Button
            className={classes.submitButton}
            color="primary"
            variant={"contained"}
          >
            Click to Select
          </Button>
        </div>
      )}
      {error && (
        <Typography variant="h6" className={classes.error}>
          {error}
        </Typography>
      )}
      <Button
        className={classes.button}
        color="primary"
        variant={"contained"}
        onClick={onSubmit}
        disabled={!images.length}
      >
        Submit
      </Button>
    </div>
  );
};

Step3.propTypes = {
  data: PropTypes.object,
};

export default Step3;
