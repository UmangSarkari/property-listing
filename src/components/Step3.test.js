import { render } from "@testing-library/react";
import Step3 from "./Step3";

test("Renders Step2 Correctly", () => {
  const data = {
    address: "Sothe Africa",
    bedroom: 4,
    bathroom: 2,
    description: "Lucky you to get the optional description here.",
  };
  const { asFragment } = render(<Step3 data={data} />);
  expect(asFragment()).toMatchSnapshot();
});
