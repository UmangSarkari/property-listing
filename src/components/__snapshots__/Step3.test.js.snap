// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Step2 Correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-step3-4"
    id="step3"
  >
    <section
      class="makeStyles-list-7"
    />
    <div
      class="makeStyles-dragDrop-5"
      role="presentation"
      tabindex="0"
    >
      <input
        accept="image/png,.png,.jpg,.jpeg,image/jpg,.png,.jpg,.jpeg"
        multiple=""
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <p>
        Drag 'n' drop images here or Click below to select
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitButton-1 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Click to Select
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Submit
      </span>
    </button>
  </div>
</DocumentFragment>
`;
